packages/contracts/src/chugsplash/config.ts

```diff
@@ -0,0 +1,143 @@
+/* External Imports */
+import hre from 'hardhat'
+import { cloneDeep, isPlainObject } from 'lodash'
+import { ethers } from 'ethers'
+
+type SolidityVariable =
+  | string
+  | number
+  | Array<SolidityVariable>
+  | {
+      [name: string]: SolidityVariable
+    }
+
+export interface ChugSplashConfig {
+  contracts: {
+    [name: string]: {
+      address?: string
+      source: string
+      variables?: {
+        [name: string]: SolidityVariable
+      }
+    }
+  }
+}
+
+/**
+ * Parses any template strings found inside of a variable. Will parse recursively if the variable
+ * is an array or a plain object. Keys inside plain objects can also be templated in.
+ * @param variable Variable to replace.
+ * @param env Environment variables to inject into {{ env.X }} template strings.
+ * @param addresses Contract addresses to inject into {{ contract.X }} template strings.
+ * @returns Modified variable with template strings replaced.
+ */
+const parseVariable = (
```

Comment:
This needs unit tests.  It is also possible to use templating libraries that have unit tests